### PR TITLE
docs(zh-hant): introducing jsx

### PR DESCRIPTION
**Pull request by itszero**
This one is pretty hard because there're a lot of overloaded terms: React elements v.s. HTML elements etc. I try to stick to the glossary as to what to translate and what to keep in English.

Let me know if I missed anything or any feedbacks! :smile:

### CODE REVIEW

File: content/docs/introducing-jsx.md [hunk 1]
```diff
@@ -128,28 +129,28 @@ const element = (
...
+在 JSX 之中可以安全的直接嵌入使用者輸入：
 
 ```js
 const title = response.potentiallyMaliciousInput;
 // This is safe:
 const element = <h1>{title}</h1>;
 ```
 
-By default, React DOM [escapes](http://stackoverflow.com/questions/7381974/which-characters-need-to-be-escaped-on-html) any values embedded in JSX before rendering them. Thus it ensures that you can never inject anything that's not explicitly written in your application. Everything is converted to a string before being rendered. This helps prevent [XSS (cross-site-scripting)](https://en.wikipedia.org/wiki/Cross-site_scripting) attacks.
+React DOM 預設會在 render 之前 [escape](http://stackoverflow.com/questions/7381974/which-characters-need-to-be-escaped-on-html) 所有嵌入在 JSX 中的變數。這保證你永遠不會不小心注入任何不是直接寫在你的應用程式中的東西。所有變數都會在 render 之前轉為字串，這可以避免 [XSS (跨網站指令碼)](https://en.wikipedia.org/wiki/Cross-site_scripting) 攻擊。
```
Comment (bugtender) [inline content/docs/introducing-jsx.md]:
```suggestion
React DOM 預設會在 render 之前 [escape](http://stackoverflow.com/questions/7381974/which-characters-need-to-be-escaped-on-html) 所有嵌入在 JSX 中的變數。這保證你永遠不會不小心注入任何不是直接寫在你的應用程式中的東西。所有變數都會在 render 之前轉為字串，這可以避免 [XSS（跨網站指令碼）](https://en.wikipedia.org/wiki/Cross-site_scripting)攻擊。
```

File: content/docs/introducing-jsx.md [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
 
-### Why JSX? {#why-jsx}
+### 為什麼要用 JSX? {#why-jsx}
 
-React embraces the fact that rendering logic is inherently coupled with other UI logic: how events are handled, how the state changes over time, and how the data is prepared for display.
+React 擁抱了 render 邏輯從根本上就得跟其他 UI 邏輯綁在一起的事實：事件要怎麼處理？隨著時間經過 state 會如何變化？以及要怎麼將資料準備好用於顯示？
 
-Instead of artificially separating *technologies* by putting markup and logic in separate files, React [separates *concerns*](https://en.wikipedia.org/wiki/Separation_of_concerns) with loosely coupled units called "components" that contain both. We will come back to components in a [further section](/docs/components-and-props.html), but if you're not yet comfortable putting markup in JS, [this talk](https://www.youtube.com/watch?v=x7cQ3mrcKaY) might convince you otherwise.
+與其刻意的將*技術*拆開，把標籤語法跟邏輯拆放於不同檔案之中，React [將*焦點*分散](https://en.wikipedia.org/wiki/Separation_of_concerns)的方法是將其拆分為很多同時包含 UI 與邏輯的 component，而彼此之間很少互相依賴。我們會在[之後的章節](/docs/components-and-props.html)中回來探討 component 這個主題，但如果你還沒被說服接受將標籤語法寫在 JS 裡頭，[這個演講](https://www.youtube.com/watch?v=x7cQ3mrcKaY)或許會說服你。
```
[the remark this answers is not in the document]
Comment (itszero) [reply content/docs/introducing-jsx.md]: should we use bold instead? The default typography settings are also making it looks very bad on Linux, as in italic characters would clash into its neighbours.

File: content/docs/introducing-jsx.md [hunk 3]
```diff
@@ -175,10 +176,10 @@ const element = {
...
 ```
 
-These objects are called "React elements". You can think of them as descriptions of what you want to see on the screen. React reads these objects and uses them to construct the DOM and keep it up to date.
+這種物件被稱呼為 "React element"。你可以想像他們描述的是你想要在螢幕上看到的東西，React 會讀取這些物件並用這些描述來產生 DOM 並保持他們在最新狀態。
 
-We will explore rendering React elements to the DOM in the next section.
+我們會在下一個章節探討如何把 React element render 到 DOM 之中。
 
->**Tip:**
+>**提示:**
```
Comment (chentsulin) [inline content/docs/introducing-jsx.md]: `:` -> `：`

File: content/docs/introducing-jsx.md [hunk 4]
```diff
@@ -87,37 +86,39 @@ function getGreeting(user) {
...
 ```js
 const element = <img src={user.avatarUrl}></img>;
 ```
 
+不要在嵌入 JavaScript expression 作為屬性的時候同時使用引號或是大括號。你應該要在使用字串屬性的時候使用引號，使用 expressions 的時候使用大括號，但不要同時使用。
+
 Don't put quotes around curly braces when embedding a JavaScript expression in an attribute. You should either use quotes (for string values) or curly braces (for expressions), but not both in the same attribute.
 
->**Warning:**
+>**注意:**
```
Comment (chentsulin) [inline content/docs/introducing-jsx.md]: `:` -> `：`

File: content/docs/introducing-jsx.md [hunk 5]
```diff
@@ -128,28 +129,28 @@ const element = (
...
 
-### JSX Prevents Injection Attacks {#jsx-prevents-injection-attacks}
+### JSX 防範注入攻擊 {#jsx-prevents-injection-attacks}
 
-It is safe to embed user input in JSX:
+在 JSX 之中可以安全的直接嵌入使用者輸入：
 
 ```js
 const title = response.potentiallyMaliciousInput;
 // This is safe:
```
Comment (chloewlin) [inline content/docs/introducing-jsx.md]:
```suggestion
// 這是安全的：
```

File: content/docs/introducing-jsx.md [hunk 6]
```diff
@@ -158,11 +159,11 @@ const element = (
...
-  'Hello, world!'
+  '你好，世界!'
 );
 ```
 
-`React.createElement()` performs a few checks to help you write bug-free code but essentially it creates an object like this:
+`React.createElement()` 會進行一些檢查以幫助你寫出沒有 bug 的程式，但基本上它會產生類似下面的物件：
 
 ```js
 // Note: this structure is simplified
```
Comment (chloewlin) [inline content/docs/introducing-jsx.md]:
```suggestion
// 注意：這是被簡化的結構
```

File: content/docs/introducing-jsx.md [hunk 7]
```diff
@@ -87,37 +86,39 @@ function getGreeting(user) {
...
-You may also use curly braces to embed a JavaScript expression in an attribute:
+你也可以在屬性中使用大括號來嵌入一個 JavaScript expression：
 
 ```js
 const element = <img src={user.avatarUrl}></img>;
 ```
 
+不要在嵌入 JavaScript expression 作為屬性的時候同時使用引號或是大括號。你應該要在使用字串屬性的時候使用引號，使用 expressions 的時候使用大括號，但不要同時使用。
+
 Don't put quotes around curly braces when embedding a JavaScript expression in an attribute. You should either use quotes (for string values) or curly braces (for expressions), but not both in the same attribute.
```
Comment (chloewlin) [inline content/docs/introducing-jsx.md]:
```suggestion
```
You forgot to delete this passage. lol

**Comment by chloewlin** [comment]
@itszero I spotted three small things. Once they're fixed, I will merge this in asap.

**Comment by itszero** [comment]
Updated!

File: content/docs/introducing-jsx.md [hunk 8]
```diff
@@ -42,18 +42,18 @@ ReactDOM.render(
...
 function formatName(user) {
-  return user.firstName + ' ' + user.lastName;
+  return user.lastName + ' ' + user.firstName;
 }
 
 const user = {
-  firstName: 'Harper',
-  lastName: 'Perez'
+  firstName: '小明',
+  lastName: '王'
```
Comment (chloewlin) [inline content/docs/introducing-jsx.md]: @itszero perhaps we can change this back to English?

File: content/docs/introducing-jsx.md [hunk 9]
```diff
@@ -158,14 +159,14 @@ const element = (
...
+  'Hello, World!'
 );
 ```
 
-`React.createElement()` performs a few checks to help you write bug-free code but essentially it creates an object like this:
+`React.createElement()` 會進行一些檢查以幫助你寫出沒有 bug 的程式，但基本上它會產生類似下面的物件：
 
 ```js
-// Note: this structure is simplified
+// // 注意：這是簡化過的結構
```
Comment (chloewlin) [inline content/docs/introducing-jsx.md]:
```suggestion
// 注意：這是簡化過的結構
```

File: content/docs/introducing-jsx.md (hunk 7 above)
Comment (chloewlin) [inline content/docs/introducing-jsx.md]:
```suggestion
```
For some reason I am still seeing line 105. Was it deleted? lol

**Comment by itszero** [comment]
:man_facepalming: tripple checked it is removed this time!

File: content/docs/introducing-jsx.md [hunk 10]
```diff
@@ -1,36 +1,36 @@
...
 ```
 
-This funny tag syntax is neither a string nor HTML.
+這個有趣的標籤語法不是一個字串也不是 HTML。
 
-It is called JSX, and it is a syntax extension to JavaScript. We recommend using it with React to describe what the UI should look like. JSX may remind you of a template language, but it comes with the full power of JavaScript.
+這個語法叫做 JSX，是一個 JavaScript 的語法擴充。我們推薦你在寫 React 的時候透過這個語法來描述使用者界面的外觀。 JSX 可能為讓你想到一些樣板語言，但不一樣的地方是 JSX 允許你使用 JavaScript 所有的功能。
 
-JSX produces React "elements". We will explore rendering them to the DOM in the [next section](/docs/rendering-elements.html). Below, you can find the basics of JSX necessary to get you started.
+執行 JSX 會產生 React "element"。我們會在[下一個章節](/docs/rendering-elements.html)深入如何將這些輸出 render 到 DOM 裡頭。接下來，我們將帶您了解 JSX 的基礎。
```
Comment (chloewlin) [inline content/docs/introducing-jsx.md]:
```suggestion
執行 JSX 會產生 React「element」。我們會在[下一個章節](/docs/rendering-elements.html)深入如何將這些輸出 render 到 DOM 裡頭。接下來，我們將帶您了解 JSX 的基礎。
```

File: content/docs/introducing-jsx.md [hunk 11]
```diff
@@ -175,10 +174,10 @@ const element = {
 };
 ```
 
-These objects are called "React elements". You can think of them as descriptions of what you want to see on the screen. React reads these objects and uses them to construct the DOM and keep it up to date.
+這種物件被稱呼為 "React element"。你可以想像他們描述的是你想要在螢幕上看到的東西，React 會讀取這些物件並用這些描述來產生 DOM 並保持他們在最新狀態。
```
Comment (chloewlin) [inline content/docs/introducing-jsx.md]:
```suggestion
這種物件被稱呼為「React element」。你可以想像他們描述的是你想要在螢幕上看到的東西，React 會讀取這些物件並用這些描述來產生 DOM 並保持他們在最新狀態。
```

" " => 「」

**Comment by chloewlin** [comment]
Two small changes, then we're done! 💯

File: content/docs/introducing-jsx.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (chentsulin) [reply content/docs/introducing-jsx.md]: `Separation of concerns` usually be translated to `關注點分離`. (can't get any relevant result when searching `焦點分散` on Google)

**Pull request merged**